File: bindings/javascript/Polycode/RenderBuffer.js
function RenderBuffer(width,height,attachDepthBuffer,floatingPoint) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.RenderBuffer(width,height,attachDepthBuffer,floatingPoint)
	}
	Object.defineProperties(this, {
		'platformData': { enumerable: true, configurable: true, get: RenderBuffer.prototype.__get_platformData, set: RenderBuffer.prototype.__set_platformData},
		'depthBufferPlatformData': { enumerable: true, configurable: true, get: RenderBuffer.prototype.__get_depthBufferPlatformData, set: RenderBuffer.prototype.__set_depthBufferPlatformData}
	})
}


RenderBuffer.prototype.__get_platformData = function() {
	var retVal = new RendererPlatformData("__skip_ptr__")
	retVal.__ptr = 	Polycode.RenderBuffer__get_platformData(this.__ptr)
	return retVal
}

RenderBuffer.prototype.__set_platformData = function(val) {
	Polycode.RenderBuffer__set_platformData(this.__ptr, val.__ptr)
}

RenderBuffer.prototype.__get_depthBufferPlatformData = function() {
	var retVal = new RendererPlatformData("__skip_ptr__")
	retVal.__ptr = 	Polycode.RenderBuffer__get_depthBufferPlatformData(this.__ptr)
	return retVal
}

RenderBuffer.prototype.__set_depthBufferPlatformData = function(val) {
	Polycode.RenderBuffer__set_depthBufferPlatformData(this.__ptr, val.__ptr)
}

Duktape.fin(RenderBuffer.prototype, function (x) {
	if (x === RenderBuffer.prototype) {
		return;
	}
	Polycode.RenderBuffer__delete(x.__ptr)
})

RenderBuffer.prototype.getWidth = function() {
	return Polycode.RenderBuffer_getWidth(this.__ptr)
}

RenderBuffer.prototype.getHeight = function() {
	return Polycode.RenderBuffer_getHeight(this.__ptr)
}

File: bindings/lua/Polycode/RenderBuffer.lua
class "RenderBuffer"


function RenderBuffer:__getvar(name)
	if name == "platformData" then
		local retVal = Polycode.RenderBuffer_get_platformData(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["RendererPlatformData"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "depthBufferPlatformData" then
		local retVal = Polycode.RenderBuffer_get_depthBufferPlatformData(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["RendererPlatformData"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end

function RenderBuffer:__setvar(name,value)
	if name == "platformData" then
		Polycode.RenderBuffer_set_platformData(self.__ptr, value.__ptr)
		return true
	elseif name == "depthBufferPlatformData" then
		Polycode.RenderBuffer_set_depthBufferPlatformData(self.__ptr, value.__ptr)
		return true
	end
	return false
end
function RenderBuffer:RenderBuffer(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.RenderBuffer(unpack(arg))
	end
end

function RenderBuffer:getWidth()
	local retVal =  Polycode.RenderBuffer_getWidth(self.__ptr)
	return retVal
end

function RenderBuffer:getHeight()
	local retVal =  Polycode.RenderBuffer_getHeight(self.__ptr)
	return retVal
end

function RenderBuffer:__delete()
	if self then Polycode.delete_RenderBuffer(self.__ptr) end
end
